core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -90,6 +90,10 @@ class LogManager(logDirs: Seq[File],
   private val _liveLogDirs: ConcurrentLinkedQueue[File] = createAndValidateLogDirs(logDirs, initialOfflineDirs)
   @volatile private var _currentDefaultConfig = initialDefaultConfig
   @volatile private var numRecoveryThreadsPerDataDir = recoveryThreadsPerDataDir
+  // log dir path -> number of Remaining logs map for remainingLogsToRecover metric
+  private val numRemainingLogs: ConcurrentMap[String, Int] = new ConcurrentHashMap[String, Int]
+  // log recovery thread name -> number of remaining segments map for remainingSegmentsToRecover metric
+  private val numRemainingSegments: ConcurrentMap[String, Int] = new ConcurrentHashMap[String, Int]
```

Comment:
Could we make these two maps local in loadLogs()?

Reply:
Done